src/chain_fusion_backend/src/job/submit_result.rs

```diff
@@ -12,22 +13,23 @@ pub async fn submit_result(result: String, job_id: U256) {
     //TODO: Should probably be hardcoded. Recomputing the hash every time is unnecessary
     let function_signature = "callback(string,uint256)";
 
-    // let mut data = keccak256(function_signature).as_ref()[0..4].to_vec();
-    // data.extend(ethers_core::abi::AbiEncode::encode(result));
     let selector = &keccak256(function_signature.as_bytes())[0..4];
     let args = (result, job_id).encode();
     let mut data = Vec::from(selector);
     data.extend(args);
 
+    let gas_limit = U256::from(5000000);
+    let fee_estimates = fees::estimate_transaction_fees(9).await;
```

Comment:
should we pas those to the function like we do for `transfer_eth`?

Reply:
Which function?

Refactor to `call_contract(sig, data, gas_limit)` or something?

Reply:
i meant `submit_result`. 
within both `transfer_eth` and `submit_result` we 
1. create a request
2. sign it
3. send it to the network
4. check the status 

for `transfer_eth` we pass the `gas` and `fee_estimate` to the function, for `submit_result` it's hardcoded in the body of the function. ideally both behave the same. i think the `fee_estimate` should be part of the body in both cases, so we only need to decide on how to proceed with `gas`